File: app/src/main/java/com/shop440/Login/ApiRequest.java
package com.shop440.Login;

import com.shop440.Models.User;
import com.shop440.Utils.Urls;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;

/**
 * Created by mmumene on 03/09/2017.
 */

public interface ApiRequest {
    @POST(Urls.NEW_USER)
    Call<User> createUser(@Body User user);

    @POST(Urls.LOGIN)
    Call<User> login(@Body User user);
}

File: app/src/main/java/com/shop440/features/auth/ApiRequest.kt
package com.shop440.features.auth

import com.shop440.repository.dao.models.User
import com.shop440.repository.api.response.OtpResponse
import com.shop440.repository.api.response.UserResponse
import com.shop440.repository.api.Urls

import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query

/**
 * Created by mmumene on 03/09/2017.
 */

interface ApiRequest {
    @POST(Urls.NEW_USER)
    fun createUser(@Body user: User): Call<UserResponse>

    @POST(Urls.LOGIN)
    fun login(@Body user: User): Call<UserResponse>

    @GET(Urls.OTP)
    fun requestOtp(@Query("p") query: String): Call<OtpResponse>

    @POST(Urls.CHECKPHONE)
    fun checkAvailability(@Body user: User) : Call<User>
}
